transport/src/main/java/io/netty/channel/nio/AbstractNioByteChannel.java

```diff
@@ -157,12 +158,71 @@ public final void read() {
         }
     }
 
+    private enum WriteStatus {
+        SET_OP_WRITE, // We have not written out the whole message and output channel is not writable
+        INCOMPLETE, // We have not written out the whole message after the spin count
+        DONE // we have written out the whole message
+    }
+
+    // returns whether we should setOpWrite
+    private WriteStatus doWriteFileRegion(ChannelOutboundBuffer in, FileRegion region,
+            int writeSpinCount) throws Exception {
+        if (region instanceof SliceableFileRegion) {
+            SliceableFileRegion sliceableRegion = (SliceableFileRegion) region;
+            if (!sliceableRegion.isTransferable()) {
+                return WriteStatus.DONE;
+            }
+            WriteStatus status = WriteStatus.INCOMPLETE;
+            long flushedAmount = 0;
+
+            for (int i = writeSpinCount - 1; i >= 0; i--) {
```

Comment:
Why not use a plain incrementing for loop?

Reply:
this is preexisting IIRC ... the index isn't used and we don't have to declare an additional constant for `end` this way.